configuration/azure/appconfig/appconfig.go

```diff
@@ -160,32 +160,71 @@ func parseMetadata(meta configuration.Metadata) (metadata, error) {
 func (r *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequest) (*configuration.GetResponse, error) {
 	keys := req.Keys
 	items := make([]*configuration.Item, 0, len(keys))
```

Comment:
This allocation of `items` seems to be only useful in the case where `len(keys) != 0`. Could you move it into to `else` block (to avoid allocating memory when not needed?)